src/rgw/rgw_tag.h

```diff
@@ -0,0 +1,41 @@
+#ifndef RGW_TAG_H
+#define RGW_TAG_H
+
+#include <map>
+#include <string>
+#include <include/types.h>
+
+#include "rgw_common.h"
+
+using std::string;
+using std::map;
+
+class RGWObjTags
+{
+ protected:
+  map <string, string> tags;
+ public:
+  RGWObjTags() {}
+  ~RGWObjTags() {}
+
+  void encode(bufferlist& bl) const {
+    ENCODE_START(1,1,bl);
+    ::encode(tags, bl);
+    ENCODE_FINISH(bl);
+  }
+
+  void decode(bufferlist::iterator &bl) {
+    DECODE_START_LEGACY_COMPAT_LEN(1, 1, 1, bl);
+    ::decode(tags,bl);
+    DECODE_FINISH(bl);
+  }
+
+  void dump(Formatter *f) const;
```

Comment:
it's something I'd like to see loooong term since I'd like to make people use unique_ptr<Formatter> instead of Formatter* and *f is nicer than f.get().

(That and if people wanted to use a shared_ptr<Formatter> in some case the code they call wouldn't have to care.)